### PR TITLE
[SOAR-17011] - Microsoft ATP -Adding the following as new action types to  action ['Warn', 'Block', 'Audit']

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

the customer wanted the ability to makes sure that the generate alerts flag could be disabled when creating adding an ioc to the blacklist. Please note as stated in the Microsoft docs and I have added this to the description of this new field that it will not work will all action types

image

Describe the proposed changes:

  -  Adding the following as new action types to `blacklist` action ['Warn', 'Block', 'Audit']
  -  Add a new flag in the `blacklist` action to toggle generateAlerts flag
  - Bump SDK to version 5.4.9
  - bump version of validators used for a medium snyk vuln
  
  https://rapid7.atlassian.net/browse/SOAR-17011

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

testing of all the new actions and new flag have been added to the tickets comments - https://rapid7.atlassian.net/browse/SOAR-17011?focusedCommentId=1487815

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

snyk testing 

```
❯ snyk test

Testing /Users/rbowden/insight_repos/insightconnect-plugins/plugins/microsoft_atp...

Organization:      soar-insightconnect-new
Package manager:   pip
Target file:       requirements.txt
Project name:      microsoft_atp
Open source:       no
Project path:      /Users/rbowden/insight_repos/insightconnect-plugins/plugins/microsoft_atp
Licenses:          enabled

✔ Tested 14 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.
```

### CODE REVIEW

**Review by ablakley-r7 — APPROVED** [review]

File: plugins/microsoft_atp/plugin.spec.yaml [hunk 1]
```diff
@@ -26,8 +26,48 @@ hub_tags:
   features: []
 sdk: 
   type: full 
-  version: 5 
+  version: 5.4.9 
   user: nobody
+links:
+ - "[Windows Defender ATP](https://www.microsoft.com/en-us/windowsforbusiness/windows-atp)"
+ - "[Windows Defender ATP API Start Page](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-atp/use-apis)"
+ - "[Windows Defender ATP API Endpoints](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-atp/exposed-apis-list)"
```
Comment (cmcnally-r7) [inline plugins/microsoft_atp/plugin.spec.yaml:34]: Could we remove these from links and only keep in references? No point having overlap

File: plugins/microsoft_atp/plugin.spec.yaml [hunk 2]
```diff
@@ -26,8 +26,48 @@ hub_tags:
...
+references:
+ - "[Windows Defender ATP](https://www.microsoft.com/en-us/windowsforbusiness/windows-atp)"
```
Comment (cmcnally-r7) [inline plugins/microsoft_atp/plugin.spec.yaml:36]: same here but vice versa

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

File: plugins/microsoft_atp/plugin.spec.yaml (hunk 1 above)
Comment (rbowden-r7) [reply plugins/microsoft_atp/plugin.spec.yaml]: the links and references have been updated

File: plugins/microsoft_atp/plugin.spec.yaml (hunk 2 above)
Comment (rbowden-r7) [reply plugins/microsoft_atp/plugin.spec.yaml]: the links and references have been updated

**Review by cmcnally-r7 — APPROVED** [review]

**Pull request merged**